lib/src/block_syntaxes/blockquote_syntax.dart

```diff
@@ -97,3 +97,65 @@ class BlockquoteSyntax extends BlockSyntax {
     return Element('blockquote', children);
   }
 }
+
+/// Parses GitHub Alerts blocks.
+///
+/// See also: https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
+class AlertBlockSyntax extends BlockSyntax {
+  @override
+  RegExp get pattern => alertPattern;
+
+  const AlertBlockSyntax();
+
+  @override
+  List<Line> parseChildLines(BlockParser parser) {
+    return const BlockquoteSyntax().parseChildLines(parser);
+  }
+
+  @override
+  Node parse(BlockParser parser) {
+    // Parse the alert type from the first line.
+    final type =
+        pattern.firstMatch(parser.current.content)!.group(1)!.toLowerCase();
+
+    final childLines = parseChildLines(parser);
+    // Until we've parse all the child lines, we can't actually know if this is
+    // a blockquote containing `[!note]` or if this is an alert-block.
+    //
+    // This is because `> [!note]` is not a valid alert-block!
+    final isBlockquote = childLines.length <= 1;
+
+    if (!isBlockquote) {
+      // Always remove the first line, this is the line that contained the type.
+      childLines.removeAt(0);
+    }
+
+    // Recursively parse the contents of the blockquote.
+    final children = BlockParser(childLines, parser.document).parseLines(
+      // The setext heading underline cannot be a lazy continuation line in a
+      // block quote.
+      // https://spec.commonmark.org/0.30/#example-93
+      disabledSetextHeading: BlockquoteSyntax._lazyContinuation,
+      parentSyntax: this,
+    );
+
+    if (isBlockquote) {
+      return Element('blockquote', children);
```

Comment:
This is ugly, but I don't see a better way, unless we make `BlockquoteSyntax` do everything.

I suppose we could in theory make `BlockquoteSyntax` check if `AlertBlockSyntax` is enabled by doing:`

```dart
final isAlertSyntaxEnabled = parser.document.blockSyntaxes.any(
  (s) => s is AlertBlockSyntax,
);
```

If we could give `AlertBlockSyntax` a factory singleton constructor we could do:

```dart
final isAlertSyntaxEnabled = parser.document.blockSyntaxes.contains(
  const AlertBlockSyntax(),
);
```

If inside `BlockquoteSyntax` we had `isAlertSyntaxEnabled` then it could implement both. Ofcourse this would mean that enabling `AlertBlockSyntax` wouldn't work without enabling `BlockquoteSyntax`.
I'm not sure why we should ever allow such configuration, we surely don't intent to test it, but that's how the package is structured.


---------------------

The point with implementing alert-syntax inside `BlockquoteSyntax` is that alert-syntax is really a post-processing step for blockquote. But I don't really see how `package:markdown` has the facilities to post-processing.